var/spack/repos/builtin/packages/py-virtualenv/package.py

```diff
@@ -24,31 +25,48 @@ class PyVirtualenv(PythonPackage):
     version("13.0.1", sha256="36c2cfae0f9c6462264bb19c478fc6bab3478cf0575f1027452e975a1ed84dbd")
     version("1.11.6", sha256="3e7a4c151e2ee97f51db0215bfd2a073b04a91e9786df6cb67c916f16abe04f7")
 
-    depends_on("python@2.7:2.8,3.4:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.5:", when="@20.10.0:", type=("build", "run"))
+    depends_on("python@3.7:", when="@20.18.0:", type=("build", "run"))
     depends_on("python@3.6:", when="@20.16.4:", type=("build", "run"))
+    depends_on("python@2.7:2.8,3.5:", when="@20.5.0:", type=("build", "run"))
+    depends_on("python@2.7:2.8,3.4:", type=("build", "run"))
```

Comment:
You could honestly remove all these, Spack only supports 3.7+ (soon to be 3.8+)

Reply:
Done. I am always reluctant to do so because it could result in problems when backporting to older spack releases (which I sometimes have to do).